scripts/lint.py

```diff
@@ -661,45 +661,122 @@ def lint_workspace_lints(cargo_file_content: str) -> str | None:
 
 # -----------------------------------------------------------------------------
 
+force_capitalized_words = [
+    "2D",
+    "3D",
+    "Apache",
+    "API",
+    "APIs",
+    "Arrow",
+    "Bevy",
+    "C",
+    "C++",
+    "C++17,",  # easier than coding up a special case
+    "C++17",
+    "Colab",
+    "Google",
+    "Jupyter",
+    "Linux",
+    "Mac",
+    "Numpy",
+    "nuScenes",
+    "Pixi",
+    "Python",
+    "Q1",
+    "Q2",
+    "Q3",
+    "Q4",
+    "Rerun",
+    "Rust",
+    "SAM",
+    "Viewer",
+    "Wasm",
+    "Windows",
+    "April",
+]
+
+force_capitalized_words_as_lower = [word.lower() for word in force_capitalized_words]
 
-def fix_header_casing(s: str) -> str:
-    allowed_title_words = [
-        "Apache",
-        "APIs",
-        "Arrow",
-        "C",
-        "C++",
-        "Colab",
-        "Google",
-        "Jupyter",
-        "Linux",
-        "Numpy",
-        "Pixi",
-        "Python",
-        "Rerun",
-        "Rust",
-        "Wasm",
-        "Windows",
-    ]
 
+def fix_header_casing(s: str) -> str:
     def is_acronym_or_pascal_case(s: str) -> bool:
         return sum(1 for c in s if c.isupper()) > 1
 
-    new_words = []
+    new_words: list[str] = []
+    last_punctuation = None
+    inline_code_block = False
 
     for i, word in enumerate(s.strip().split(" ")):
         if word == "":
             continue
-        if word.lower() in ("2d", "3d"):
-            word = word.upper()
-        elif is_acronym_or_pascal_case(word) or any(c in ("_", "(", ".") for c in word):
-            pass  # acroym, PascalCase, code, …
-        elif i == 0:
-            # First word:
+
+        if word.startswith("`"):
+            inline_code_block = True
+        if word.endswith("`"):
+            inline_code_block = False
+
+        if last_punctuation:
             word = word.capitalize()
-        else:
-            if word not in allowed_title_words:
+            last_punctuation = None
+        elif not inline_code_block and not word.startswith("`"):
+            try:
+                idx = force_capitalized_words_as_lower.index(word.lower())
+            except ValueError:
+                idx = None
+
+            # special case: don't capitalize "web viewer" and "VRS viewer" even though we capitalize "Viewer"
```

Comment:
that would be avoided if we only enforced "Rerun Viewer"

Reply:
that's probably the way to go, but since this script works "per word" right now it's hard. I'll have to change that!

Reply:
ended up skipping it, too much hassle